### DISCUSSION TITLE
Zeek Windows Install

### DISCUSSION THREAD

**Discussion by SpaceXElaborator** [zeek/zeek#2714] [Help] [answered]
Currently trying to figure out our next steps with the Zeek install.

Base File Path (GitHub pull)
C:\zeek

Following along with the guidelines, we installed and ran cmake.exe --build . inside of the /build directory.
No errors were given and we have a zeek.exe file. But no clue where our next step is in making it work.

We had to move some of the files from
C:\zeek\scripts\base
and move those into
C:\zeek\build\scripts\base

These files were init-bare.zeek and all subfolders that were in base.
We moved the zeek.exe file from
C:\zeek\build\src\zeek.exe
to
C:\zeek\build\scripts

And we finally got zeek.exe to return some information. But, now we are lost on where to go next with retrieving where the logs are placed, how to start log capturing, and the likes.

**Comment by timwoj** [comment, marked as answer]
First off, the Zeek support on Windows right now is fairly experimental. There's a lot of features that are wholly untested, and a lot of bugs left to fix. That said...

I highly recommend working from a bash shell of some sort (gitbash, cygwin, etc). It will make your life considerably easier as you get used to how Zeek works, and while the team works on making life on Windows a little easier. From the build directory, you can run `source zeek-path-dev.sh`. That will add a few variables to your environment, two of which are `ZEEKPATH` and `ZEEK_PLUGIN_PATH`. `ZEEKPATH` is a list of paths that zeek will search for scripts at startup, which will help it find things like `init-bare.zeek`. `ZEEK_PLUGIN_PATH` is a list of paths that zeek will search for externally-built plugins, which is slightly less important on Windows since we haven't spent a lot of time on plugin support there yet. If you don't want to run full-time from bash, you can look at what the script sets those to and just set them in your environment directly. I do that when I'm running in a debugger, since the debugger can set everything up for me automatically.

As for live packet capture, you'll need to build against npcap (https://npcap.com). The default libpcap works fine for reading pcap files, but they never fixed support for network interfaces after Microsoft changed things a while back (Windows 7? 8?). To build against that, you can pass `-DPCAP_ROOT_DIR=<path to npcap sdk>` when running cmake.

Logs are automatically stored in whatever directory you start Zeek from. For the rest, https://docs.zeek.org/en/master/quickstart.html#zeek-as-a-command-line-utility is a good starting point in the documentation.
